### ISSUE TITLE
Efficient Access Patterns for Databases with Many Collections

### ISSUE THREAD

**Issue by awtkns** [supabase/vecs#17]
Currently the way vecs retrieves collections by name does not seem to scale well when working with a database with many collections (ie 1000+). 

By looking at the code this seems that vecs is fetching ALL the collections and then looping through to find one with a matching name. It would be more efficient if there was a where clause in `_list_collections` in which we could specify the collection by name so that only one would be fetched.

Maybe i'm missing something?
```python
    def get_collection(self, name: str) -> Collection:
        """Get an existing collection"""
        from vecs.collection import Collection

        collections = Collection._list_collections(self)
        for collection in collections:
            if collection.name == name:
                return collection
        raise CollectionNotFound("No collection found with requested name")
        
        # ...
        
        @classmethod
    def _list_collections(cls, client: "Client") -> List["Collection"]:
        query = text(
            """
        select
            relname as table_name,
            atttypmod as embedding_dim
        from
            pg_class pc
            join pg_attribute pa
                on pc.oid = pa.attrelid
        where
            pc.relnamespace = 'vecs'::regnamespace
            and pc.relkind = 'r'
            and pa.attname = 'vec'
            and not pc.relname ^@ '_'
        """
        )
        xc = []
        with client.Session() as sess:
            for name, dimension in sess.execute(query):
                existing_collection = cls(name, dimension, client)
                xc.append(existing_collection)
        return xc
```

**Comment by olirice** [comment]
yep I agree, that would be a nice performance tweak for users who need lots of collections.

marked as `good first issue` if you're interested in contributing!

**Comment by olirice** [comment]
resolved by #18
